test/lsVersion.ts

```diff
@@ -0,0 +1,73 @@
+import { StatusBar, VSBrowser, Workbench } from "vscode-extension-tester";
+import { expect } from "chai";
+import * as path from "path";
+import { getStatusBarItem } from "./getStatusBarItem";
+import { homedir } from "os";
+
+describe("LS version", function () {
+  this.timeout(50000);
+
+  it("Checks correct scarb precedence", async function () {
+    // asdf is in fact in PATH and in our tests it is first scarb in  path, special case this.
+    // It is caused by `@actions/core addPath` implementation.
+    // See: https://github.com/actions/toolkit/blob/01f21badd5a7522507f84558503b56c4deec5326/packages/core/src/core.ts#L107
+    if (process.env.PATH_SCARB_VERSION && process.env.ASDF_SCARB_VERSION) {
+      process.env.PATH_SCARB_VERSION = process.env.ASDF_SCARB_VERSION;
+    }
+
+    const scarbs = [
+      // Order is important here.
+      process.env.CONFIG_SCARB_VERSION,
+      process.env.PATH_SCARB_VERSION,
+      process.env.ASDF_SCARB_VERSION,
+    ];
+
+    const expectedScarbVersion = scarbs.find(Boolean); // Find first with value.
+
+    // Ignore this test locally, it is strictly designed for our CI.
+    if (!expectedScarbVersion) {
+      this.skip();
+    }
+
+    if (process.env.CONFIG_SCARB_VERSION) {
+      const workbench = new Workbench();
+
+      const settings = await workbench.openSettings();
+
+      const setting = await settings.findSetting("Scarb Path", "Cairo1");
+
+      setting.setValue(path.join(homedir(), ".local", "bin", "scarb"));
+
+      await workbench.executeCommand("Cairo: Reload workspace");
+    }
+
+    await VSBrowser.instance.openResources(path.join("ui-test", "fixtures", "empty"));
+
+    const statusBar = await VSBrowser.instance.driver.wait(getStatusBarItem, 5000);
+
+    expect(statusBar).not.undefined;
+
+    const title = await statusBar!.getAttribute(StatusBar["locators"].StatusBar.itemTitle);
+
+    const matches = /Cairo, Cairo Language\nscarb (.+) \(.+\)\ncairo: .+ \(.+\)\nsierra: .+/.exec(
```

Comment:
There are a few things that I'm not easy about in this test:
1. You check for Scarb version, not for path to the found binary. Easy trap to fall into while changing something.
2. You rely on the statusbar, which we know is not implemented ideally, so is highly probably not a good foundation for testing this.
   * It also doesn't provide aforementioned path to found binary.
   * And it forces you to extract statusbar finding logic that you did in a quirky way imo (should be in `test-support` at least).

Have you considered making a (maybe test-only) escape hatch in the extension logic that will yield us a status of all running LSes? Maybe a dedicated method like `viewAnalyzedCrates`?